pkg/providers/pricing/pricing.go

```diff
@@ -148,6 +148,9 @@ func (p *Provider) UpdateOnDemandPricing(ctx context.Context) error {
 	var onDemandPrices, onDemandMetalPrices map[string]float64
 	var onDemandErr, onDemandMetalErr error
 
+	p.mu.Lock()
```

Comment:
I'm wondering if (to avoid blocking the spot pricing update thread from running) we should just create one mutex for handling each type of data (two total), one for handling the spot pricing update and the other for the on-demand update. From what I can see, we don't access the other's data during the pricing update, so they can run completely separately from each other without blocking

Reply:
FYI: If we were going to do this, then we need to update the data that we are using to get the length of instance types on L387 since that access the on-demand pricing data within the spot pricing thread.